filip/semantics/vocabulary/target_statment.py

```diff
@@ -0,0 +1,267 @@
+
+        Returns:
+            str
+        """
+
+        if self.type == StatementType.LEAF:
+            label = self.retrieve_label(vocabulary)
+            if label == "":
+                return self.target_iri
+            return label
+        else:
+            result = "(" + self.target_statements[0].to_string(vocabulary)
+            for statement in self.target_statements[1:]:
+                result += " " + self.type + " "
+                result += statement.to_string(vocabulary) + ")"
+
+            return result
+
+    def is_fulfilled_by_iri_value(self, value: str, ancestor_values: List[str])\
+            -> bool:
+        """Test if a set of values fulfills the targetstatement;
+            Only for objectRelations
+
+        Args:
+            value (str): value to check: Class_iri of instance/individual
+            ancestor_values(List[List[str]]): List containing the ancestors iris
+                for each value (linked over index)
+        Returns:
+            bool
+        """
+
+        targets = self.get_all_targets()
+
+        values = ancestor_values
+        values.append(value)
+
+        # one sublist of targets needs to be fulfilled targets:
+        # [(a and b) or c or (d and a),....]
+        for sublist in targets:
+            sublist_fulfilled = True
+            for item in sublist:
+                if item not in values:
+                    sublist_fulfilled = False
+
+            if sublist_fulfilled:
+                return True
+        return False
+
+    def is_fulfilled_by_data_value(self, value: str, vocabulary: 'Vocabulary')\
+            -> bool:
+        """Test if a set of values fulfills the targetstatement;
+            Only for dataRelations
+
+        Args:
+            value (List[str]):  value to check
+            vocabulary (Vocabulary)
+        Returns:
+            bool
+        """
+
+        # a data target_statement theoraticly only has one statement
+        if self.target_data_value is not None:
+            return value == self.target_data_value
+
+        from .vocabulary import IdType
+        if not vocabulary.is_id_of_type(self.target_iri, IdType.datatype):
+            return False
+        else:
+            datatype = vocabulary.get_datatype(self.target_iri)
+            return datatype.value_is_valid(value)
+
+    def retrieve_label(self, vocabulary: 'Vocabulary') -> str:
+        """Get the label of the target_iri. Only logical for Leaf statements
+
+        Args:
+            vocabulary (Vocabulary): Vocabulary of the project
+
+        Returns:
+            str
+        """
+        if self.type == StatementType.LEAF:
+            if self.target_data_value is not None:
+                return self.target_data_value
+            else:
+                return vocabulary.get_label_for_entity_iri(self.target_iri)
+        return ""
+
+    def get_dependency_statements(
+            self,
+            vocabulary: 'Vocabulary',
```

Comment:
THis must be equal to this string?

Reply:
Equal to which string, could you elaborate?